src/form.js

```diff
@@ -18,27 +18,38 @@ export default class Form extends React.Component {
       question: newCardData.get('question'),
       answer: newCardData.get('answer')
     }
+    event.target.reset()
+    if (this.props.mode === 'edit') {
+      newCard.cardId = parseInt(this.props.params.cardId, 10)
+    }
+    else {
+      newCard.cardId = this.props.nextId
+    }
     this.props.submit(newCard)
     event.target.reset()
   }
+
   render() {
+    const currentCard = Object.assign({}, this.props.cards.find(card => card.cardId.toString() === this.props.params.cardId))
```

Comment:
You are making your form do too much work to figure out which card it's editing. Instead, pass it the card to edit as a prop.